R/utils-shared.R

```diff
@@ -568,6 +570,41 @@ RescaleYTop <- function(object) {
   return(object)
 }
 
+#' Add x-axis labels to five year GW summary plots, and DV hydrographs
+#' having time intervals of one year or more.
+#' @param object A gsplot, plot object.
+#' @param month_label month_label_location
+#' @param date_seq_yr
+#' @return The passed-in gsplot object, with x-axis labeled.
+XAxisLabels <- function(object, month_label, month_label_location, date_seq_yr) {
+  return(
+    mtext(
+      object,
+      text = month_label, at = month_label_location,
+      cex = 0.5, side = 1
+    ) %>%
+      mtext(
+        text = year(date_seq_yr), at = date_seq_yr + (60 * 60 * 24 * 30 * 6),
```

Comment:
Whoops, year. When it "unlists" it does that conversion, so you need to use do.call and apply the function `c` to the output of `lapply`, not `sapply`:

```r
yr.at <- do.call(c, lapply(year(date_seq_yr), function(y, plotDates){
    which.yr.dates <- which(year(plotDates) == y)
    med.date.each.year <- median(plotDates[which.yr.dates])
    return(med.date.each.year)
}, plotDates = plotDates))
```

If you want to know why, this stack overflow post explains it http://stackoverflow.com/questions/14449166/dates-with-lapply-and-sapply